### PR TITLE
Add new guild properties

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Discord has added these new properties a while back.

### CODE REVIEW

**Comment by MinnDevelopment** [comment]
We also have to add the new boosting properties `premium_tier` and `premium_subscription_count`. Member also has `premium_since` now.

**Comment by DV8FromTheWorld** [comment]
Do we want to use the client naming of "Boosted" or Premium?

**Comment by MinnDevelopment** [comment]
We could probably remove `retrieveVanityUrl` with this as well. The code is now a guild property it seems.

**Comment by DV8FromTheWorld** [comment]
Do we get an update event when it is added?

**Comment by MinnDevelopment** [comment]
@DV8FromTheWorld I haven't confirmed it but I would think so.

**Comment by DV8FromTheWorld** [comment]
If we get WS events for this, then let's remove the method.

Need to detail that it has been removed and replaced by the non network variant

File: src/main/java/net/dv8tion/jda/internal/handle/GuildMemberUpdateHandler.java [hunk 1]
```diff
@@ -117,6 +120,17 @@ protected Long handleInternally(DataObject content)
                                 member, oldNick));
             }
         }
+        if (content.hasKey("premium_since"))
+        {
+            long boostDate = 0;
+            if (!content.isNull("premium_since"))
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/handle/GuildMemberUpdateHandler.java]: Forgot the else here which sets it to 0 and removing the boostDate variable which is unused

**Comment by MinnDevelopment** [comment]
Reminder: Add the events to ListenerAdapter and add documentation

File: src/main/java/net/dv8tion/jda/internal/handle/GuildUpdateHandler.java [hunk 2]
```diff
@@ -99,7 +99,7 @@ protected Long handleInternally(DataObject content)
         {
             guild.setBannerId(bannerId);
         }
-        if (!Objects.equals(vanityCode, guild.getVanityUrl()))
+        if (!Objects.equals(vanityCode, guild.getVanityCode()))
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/handle/GuildUpdateHandler.java]: We need to add events for all of these new properties.

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/handle/GuildUpdateHandler.java]: I already did.

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/internal/handle/GuildUpdateHandler.java]: Ah, I was looking at an older version of the changes.

**Pull request merged**